### DISCUSSION TITLE
dBeaver fails to produce error messages when running a SQL script with errors.

### DISCUSSION THREAD

**Discussion by coleman-rik** [dbeaver/dbeaver#19955] [Q&A] [answered]
### Description

When running SQL and or plpgsql and there's an error in the script, dBeaver often won't display the error messages that PostgreSQL returns.
I often find myself having to copy/paste/run the same script in pgAdmin4 to find out what the database believes the error to be.

In these cases, dBeaver just shows the entire script that was attempted to run in the Statistics panel.
If you are extremely lucky, it *might* list something important in the bottom right of the main window pane.

### DBeaver Version

Community Edition 23.0.2

### Operating System

Windows 10

### Database and driver

PostgreSQL 15, 14, 11

### Steps to reproduce

1. write SQL/plpgsql script containing one or more errors in query window
2. try to run it
3. watch a Statistics window pop up with a copy of the script you were trying to run, perhaps with an arrow pointing at some seemingly random line.
4. rerun the script in a different product to see what the error messages that the database sent in response actually are.

**Comment by LonwoLonwo** [comment]
Hello @coleman-rik 

Please provide an example of the falling script for testing.

**Comment by LonwoLonwo** [comment]
Some footage of the issue also will be helpful.

**Comment by coleman-rik** [comment]
Sure, 
Here's the most trivial examples I can come up with.  As the SQL gets more complicated, so does the PostgreSQL error messages, as does the importance of actually knowing what they might be.

On a server there there is **no** database public run this statement:
`grant all on database public to postgres, minion, bob, stow, ditto, nagios;`

the result on dBeaver is:
![dbeaver_error_example_20230421_103705](https://user-images.githubusercontent.com/3697063/233665997-340fef1f-b1dc-4263-b91e-c14bcc86b91f.png)

the same command run against the same server in pgAdmin4 returns this:
![pgAdmin_error_example_20230421_103705](https://user-images.githubusercontent.com/3697063/233666195-43271aaf-cfe5-4b78-ace3-73a1f014c76a.png)

On a server where there *is* a database idp, but the role bob **doesn't** exit run this statement:
`grant all on database idp to bob;`

the result on dBeaver is:
![dbeaver_error_example2_20230421_103705](https://user-images.githubusercontent.com/3697063/233667667-4b8dada9-53f0-4097-9ab1-f14b402fb6d0.png)

the same command run against the same server in pgAdmin4 returns this:
![pgAdmin_error_example2_20230421_103705](https://user-images.githubusercontent.com/3697063/233668036-0cc3efae-541c-49e9-a46a-6f4f4db0a54e.png)

there is nothing in the Output tab in dBeaver, there is nothing in the lower right corner of the application window in dBeaver.  What you see in those screenshots is all that dBeaver will tell you when there's an error in your SQL statement.

I hope that helps.

**Comment by E1izabeth** [comment, marked as answer]
Maybe you acidentialy hide the left part of the error view, so it looks like on the following screenshot.
![image](https://github.com/dbeaver/dbeaver/assets/28875055/d2886626-929c-4546-b55e-24525b03bfac)
Could you try clicking on the small blue arrow there to restore the view. Usually it looks like on the following screenshot for me
![image](https://github.com/dbeaver/dbeaver/assets/28875055/6a6243e2-5379-4762-90c7-7ec7ac064d07)

Please let me know if it helps.

**Comment by coleman-rik** [comment]
dBeaver 23.0.4 now shows a message at the very bottom, that wasn't there in 23.0.2.

Hitting the blue arrow, I had no Idea it was supposed to be a button, reveals this:
![dBeaver_23 0 4_hitting _button_1st_20230511_202204](https://github.com/dbeaver/dbeaver/assets/3697063/b5b98a5a-b14c-4be1-a1bc-fb63374cacfe)

dragging it to the right revealed this:
![dBeaver_23 0 4_hitting _expanding_20230511_202204](https://github.com/dbeaver/dbeaver/assets/3697063/6b2f7b1e-30a1-4b73-b2b6-96802a22084e)

So all in all, it seems that in 23.0.4 at least there's an error message in the window frame, and now that I know that the symbol pointing to the SQL (which is just a waste of space, as the exact same SQL is right above it) can be dragged open to reveal the error message, I won't have to rerun my code in pgAdmin4 as often.

Why doesn't that error panel open by default when there's an error?  Simply reiterating the same SQL that is in the query pane doesn't help the end user and only serves to hide the error message.

thanks.

**Comment by E1izabeth** [comment]
By default, the result panel with error is extended like on your second picture.
The right-side area with the query text might be useful while running long scripts with multiple queries.

I assume the issue is solved, so I close it.

**Reply by coleman-rik** [reply]
thanks. 
unfortunately, the result panel is **not** extended by default.  by default it only shows a copy of the SQL query from above and a little blue arrow on the left apparently pointing to the copy of the SQL code.

**Reply by E1izabeth** [reply]
It shouldn't be so. Probably something went wrong for you. Could you try resetting UI settings (https://github.com/dbeaver/dbeaver/wiki/Reset-UI-settings) and check, if it helps?

**Reply by coleman-rik** [reply]
that seems to have corrected that default behavior, 
thanks.
